.github/workflows/go.yml

```diff
@@ -1,30 +1,37 @@
+---
 name: Test
-on:
-- pull_request
+"on":
+  - pull_request
 
 jobs:
   test:
     strategy:
       matrix:
         go:
-        - '1.17.x'
-        - '1.18.x'
+          - 'oldstable'
+          - 'stable'
         label:
-        - [self-hosted, linux, arm64, segment]
-        - ubuntu-latest
+          - [self-hosted, linux, arm64, segment]
+          - ubuntu-latest
 
     runs-on: ${{ matrix.label }}
 
     steps:
-    - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
-    - name: Setup Go ${{ matrix.go }}
-      uses: actions/setup-go@v2
-      with:
-        go-version: ${{ matrix.go }}
+      - name: Setup Go (${{ matrix.go }})
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ matrix.go }}
 
-    - name: Download Dependencies
-      run: go mod download
+      - name: Identify OS
+        run: uname -a
 
-    - name: Run Tests
-      run: go test -race -tags=${{ matrix.tags }} ./...
+      - name: Identify Go Version
+        run: go version
```

Comment:
These start a new container per command so there's a fair amount of overhead. How about e.g.

```yml
name: Identify Host 
run: |
    uname -a
    go version
```

Reply:
For BuildKite that'd definitely be the case, but for Github Actions, it looks like jobs create their own container/VM/whatever, but steps are just separate OS processes in that same container.

As such, steps look like they're no more expensive than running separate commands in a single shell. By making them separate steps, they do create nice UI sections though, and the above info was useful for debugging the especially stale workflow config.